File: problems/121/solution.py
# Accepted: 792ms (82.86%), 28.32MB (6.24%)

class Solution:
    def maxProfit(self, prices: List[int]) -> int:
        minPrice = prices[0]
        maxProfitVal = 0
        for price in prices:
            minPrice = min(minPrice, price)
            maxProfitVal = max(maxProfitVal, price - minPrice)

        return maxProfitVal

File: problems/121/solution.cpp
// Accepted: 82ms (95.38%), 93.75MB (29.48%)

class Solution {
public:
    int maxProfit(vector<int>& prices) {
        int minPrice = prices[0];
        int maxProfitVal = 0;
        for (int price : prices) {
            minPrice = min(minPrice, price);
            maxProfitVal = max(maxProfitVal, price - minPrice);
        }

        return maxProfitVal;
    }
};
